### ISSUE TITLE
False positive when catching and rethrowing exception

### ISSUE THREAD

**Issue by moufmouf** [thecodingmachine/phpstan-strict-rules#32]
```php
try {
    //...
} catch (\Exception $e) {
    // ...
    throw $e;
}
```

This code is legit, but will throw this error:

> Do not throw the \Exception base class. Instead, extend the \Exception base class
